lua/telescope/builtin.lua

```diff
@@ -504,37 +493,18 @@ builtin.help_tags = function(opts)
     -- TODO: previewer for Vim help
     previewer = previewers.help.new(opts),
     sorter = conf.generic_sorter(opts),
-    attach_mappings = function(prompt_bufnr, map)
-      local open = function(cmd)
-        local selection = actions.get_selected_entry(prompt_bufnr)
-
+    attach_mappings = function(prompt_bufnr)
+      actions._goto_file_selection:replace(function(_, cmd)
+        local selection = actions.get_selected_entry()
         actions.close(prompt_bufnr)
-        vim.cmd(cmd .. selection.value)
-      end
-      local nhelp = function()
-        return open("help ")
-      end
-      local vhelp = function()
-        return open("vert bo help ")
-      end
-      local hhelp = function()
-        return open("help ")
-        -- Not sure how explictly make horizontal
-      end
-      local thelp = function()
-        return open("tab help ")
-      end
-      -- Perhaps it would be a good idea to have vsplit,tab,hsplit open
-      -- a builtin action that accepts a command to be ran before creating
-      -- the split or tab
-      map('i', '<CR>',  nhelp)
-      map('n', '<CR>',  nhelp)
-      map('i', '<C-v>', vhelp)
-      map('n', '<C-v>', vhelp)
-      map('i', '<C-x>', hhelp)
-      map('n', '<C-x>', hhelp)
-      map('i', '<C-t>', thelp)
-      map('n', '<C-t>', thelp)
+        if cmd == 'edit' or cmd == 'new' then
+          vim.cmd('help ' .. selection.value)
+        elseif cmd == 'vnew' then
+          vim.cmd('vert bo help ' .. selection.value)
+        elseif cmd == 'tabedit' then
+          vim.cmd('tab help ' .. selection.value)
```

Comment:
I think this can be done even better.
But this new api is amazing. So much cleaner